### DISCUSSION TITLE
为什么v5版本移除 domain_set 和 ip_set v2ray 解包的功能？

### DISCUSSION THREAD

**Discussion by daiyeqi** [IrineSistiana/mosdns#584] [General]
能够直接使用 v2ray 的数据源一直感觉很方便，希望将该功能重新加回来。或者能支持 [clash-rules](https://github.com/Loyalsoldier/clash-rules) 格式的规则。

**Comment by ghost** [comment]
.

**Reply by daiyeqi** [reply]
> https://github.com/urlesistiana/v2dat

谢谢，但还是感觉直接在数据源里引用v2ray格式的文件比较方便

**Reply by chisaato** [reply]
看 v4 文档说的是因为完整解包带来的性能问题
https://irine-sistiana.gitbook.io/mosdns-wiki/mosdns-v4/cha-jian-ji-qi-can-shu#guan-yu-v2ray-shu-ju-bao

但也有很多拥有足够性能的 x86 设备,我觉得没必要砍掉这个功能

**Reply by christaikobo** [reply]
有啥项目可以替代geosite么
说句实话2个程序完成一个程序本来可以做的事情，感觉很不爽

**Reply by chisaato** [reply]
我写了个脚本用v2dat拆包,然后v2dat和mosdns打包在同一个容器里. 只要执行脚本更新就很方便.

**Reply by christaikobo** [reply]
我知道写个脚本可以自动化这些事情，可是不爽啊！

**Reply by Zerorigin** [reply]
> 看 v4 文档说的是因为完整解包带来的性能问题
> https://irine-sistiana.gitbook.io/mosdns-wiki/mosdns-v4/cha-jian-ji-qi-can-shu#guan-yu-v2ray-shu-ju-bao
> 
> 但也有很多拥有足够性能的 x86 设备,我觉得没必要砍掉这个功能

对的，感觉有点舍大取小。
而且 Redis 缓存也砍了，从没在内存缓存开 lazy cache，用的是 Redis 的乐观缓存+ lazy cache，Redis 的整体缓存能有40％左右的命中率(配合内存缓存能有80％以上的命中率，最高能接近90%)，还是很可观的，而且打开网页啥的也基本没遇到过问题。
而且用 Redis 缓存重启也方便阿，内存缓存 dump 不就只是模仿 Redis 做一套现成的么？
都不知道功能来回+-+-的，是怎么想的。

**Comment by urlesistiana** [comment]
反对数据碎片化。不会支持其他数据格式。mosdns 只支持最原始的数据格式。

主流数据源数据都是 cidr 表，域名表。现在一个软件一个专有打包格式。到最后数据源都一样，数据包却都不通用。

**Comment by wangmice** [comment]
软件的对象毕竟是用户，加入这功能是方便更多的人，也降低使用门槛，如果大部分用户希望加入这个功能，希望作者能考虑一下。

**Comment by wy16W2pIilK1xgqN** [comment]
数据格式还是次要问题，关键是不能动态载入数据了。
总不能一更新数据就重启软件吧。

**Reply by urlesistiana** [reply]
> 总不能一更新数据就重启软件吧。

启动耗时不到0.1s。又不是要求 sla 99后面9个9，有什么不能重启的....

**Reply by wy16W2pIilK1xgqN** [reply]
这么说确实是的，但中断服务总归是不好的，特别是v4版本支持，现在反而退步了。

**Reply by urlesistiana** [reply]
v4那个功能依赖文件系统。并不是所有设备和系统都能用，而且玄学问题不少，比如只读取了一半数据。不然v5也不会去掉。稳定性第一。

**Reply by wy16W2pIilK1xgqN** [reply]
问题在文件监控吗？
我想到一个方法：
mosdns不监控文件变化，通过由用户通过特定的DNS查询，来触发数据重新载入。
比如在配置文件中设定，查询 abc.123 域名，或者在 打开一个特定 的 mosdns HTTP服务地址 /dns-query/abc.
这样来触发数据重载。

这种方案是否可行😁

**Reply by Zerorigin** [reply]
> 问题在文件监控吗？ 我想到一个方法： mosdns不监控文件变化，通过由用户通过特定的DNS查询，来触发数据重新载入。 比如在配置文件中设定，查询 abc.123 域名，或者在 打开一个特定 的 mosdns HTTP服务地址 /dns-query/abc. 这样来触发数据重载。
> 
> 这种方案是否可行😁

完全可行，你这变通能力不错阿。
就是看开发者给不给实现了。

**Reply by urlesistiana** [reply]
可行。而且考虑过。结论是和 restart mosdns 差不多...

**Reply by 4thanks** [reply]
> 可行。而且考虑过。结论是和 restart mosdns 差不多...

restart不丢内存缓存的cache吗？我用reload会丢。

**Comment by Rirmach** [comment]
没必要陷入所谓“标准化”的误区，用户用的多自然就是标准。v2的用户应该挺多的，CoreDNS都能支持maxmind的geoip，mosdns不至于还容不下一个v2。而且能来用这个的也不在乎那点性能损失，要性能有smartdns，要功能有CoreDNS，别到时候mosdns夹在中间成了四不像。
